src/lib/x509/x509_builder.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+* (C) 2025 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/x509_builder.h>
+
+#include <botan/assert.h>
+#include <botan/pubkey.h>
+#include <botan/x509_ca.h>
+#include <botan/x509_ext.h>
+
+namespace Botan {
+
+class CertificateParametersBuilder::State final {
+   public:
+      const X509_DN& subject_dn() const { return m_subject_dn; }
+
+      Extensions finalize_extensions(const Public_Key& key) const {
+         auto extensions = m_extensions;
+
+         extensions.replace(setup_alt_name(extensions));
+
+         extensions.add_new(
+            std::make_unique<Cert_Extension::Basic_Constraints>(m_is_ca_request, m_path_limit.value_or(0)), true);
+
+         const Key_Constraints usage = this->usage();
+
+         if(!usage.empty()) {
+            if(!usage.compatible_with(key)) {
+               throw Invalid_Argument("The requested key usage is incompatible with the algorithm");
+            }
+
+            extensions.add_new(std::make_unique<Cert_Extension::Key_Usage>(usage), true);
+         }
+
+         if(!m_ext_usage.empty()) {
+            extensions.add_new(std::make_unique<Cert_Extension::Extended_Key_Usage>(m_ext_usage));
+         }
+
+         return extensions;
+      }
+
+      Key_Constraints usage() const {
+         if(m_is_ca_request) {
+            return Key_Constraints::ca_constraints();
+         } else {
+            return m_usage;
+         }
+      }
+
+      void add_common_name(std::string_view cn) { add_subject_dn("X520.CommonName", cn); }
+
+      void add_country(std::string_view country) { add_subject_dn("X520.Country", country); }
+
+      void add_state(std::string_view state) { add_subject_dn("X520.State", state); }
+
+      void add_locality(std::string_view locality) { add_subject_dn("X520.Locality", locality); }
+
+      void add_serial_number(std::string_view sn) { add_subject_dn("X520.SerialNumber", sn); }
+
+      void add_org(std::string_view org) { add_subject_dn("X520.Organization", org); }
```

Comment:
Everything but this has the same name, so I'd propose changing this to `add_organization` too